include/api.php

```diff
@@ -6205,47 +6206,39 @@ function api_friendica_profile_show($type)
 
 	// get data of the specified profile id or all profiles of the user if not specified
 	if ($profile_id != 0) {
-		$r = q(
-			"SELECT * FROM `profile` WHERE `uid` = %d AND `id` = %d",
-			intval(api_user()),
-			intval($profile_id)
-		);
-
+		$r = Profile::getById(api_user(), $profile_id);
 		// error message if specified gid is not in database
 		if (!DBA::isResult($r)) {
 			throw new BadRequestException("profile_id not available");
 		}
 	} else {
-		$r = q(
-			"SELECT * FROM `profile` WHERE `uid` = %d",
-			intval(api_user())
-		);
+		$r = Profile::getListByUser(api_user());
 	}
 	// loop through all returned profiles and retrieve data and users
 	$k = 0;
 	$profiles = [];
-	foreach ($r as $rr) {
-		$profile = api_format_items_profiles($rr);
+	if (DBA::isResult($r)) {
+		foreach ($r as $rr) {
+			$profile = api_format_items_profiles($rr);
 
-		// select all users from contact table, loop and prepare standard return for user data
-		$users = [];
-		$nurls = q(
-			"SELECT `id`, `nurl` FROM `contact` WHERE `uid`= %d AND `profile-id` = %d",
-			intval(api_user()),
-			intval($rr['id'])
-		);
+			// select all users from contact table, loop and prepare standard return for user data
+			$users = [];
+			$nurls = Contact::select(['id', 'nurl'], ['uid' => api_user(), 'profile-id' => $rr['id']]);
```

Comment:
The function ```Contact::select(``` doesn't exist anymore.

Reply:
See https://github.com/friendica/friendica/pull/7437

Reply:
See https://github.com/friendica/friendica/pull/7437

Reply:
Ah, you had been faster than me :-)